xarray/backends/netCDF4_.py

```diff
@@ -115,13 +118,6 @@ def _nc4_group(ds, group, mode):
         return ds
 
 
-def _ensure_fill_value_valid(data, attributes):
```

Comment:
To clarify, it looks like I removed that code because it didn't appear to be necessary, which in hindsight was unwise-- apologies about that poor choice because I should have left the code where it was initially.  Assuming I didn't introduce an error we should be able to squash and merge, correct?